packages/app/features/home/TokenDetails.tsx

```diff
@@ -15,14 +15,21 @@ const TokenDetails = ({ tokenAddress }: { tokenAddress: `0x${string}` | undefine
     chainId: baseMainnet.id,
   })
 
-  // @TODO: add balance error check
   if (balance) {
+    if (balance.isError) {
+      throw new Error(`Error occurred while fetching balance. ${balance.error.message}`)
```

Comment:
You're misunderstanding. If we throw an error here the UI would be unresponsive and it would not be handled since we don't have any error boundaries setup. https://www.npmjs.com/package/react-error-boundary

I meant , if an error occurred, show it as a paragraph.

```suggestion
return	(
	<Paragraph fontSize={'$9'} fontWeight={'500'} color={'$error'}>
		Error occurred while fetching balance. {balance.error.message}
</Paragraph>
)
```

Reply:
😅